Public/Sdk/SelfHost/BuildXL/Tools/QTest/Tool.QTestRunner.dsc

```diff
@@ -117,18 +117,24 @@ export function runQTest(args: QTestArguments): Result {
     qTestDirToDeploy = qTestDirToDeploy || args.qTestDirToDeploy;
 
     // Microsoft internal cloud service use only
-    let qTestContextInfo = Environment.hasVariable("[Sdk.BuildXL]qtestContextInfo") ? f`${Environment.getFileValue("[Sdk.BuildXL]qtestContextInfo")}` : undefined;
-    let untrackingCBPaths =  Environment.hasVariable("[Sdk.BuildXL]qtestContextInfo") ? {
-        unsafe: {
-            untrackedPaths: [
-                qTestContextInfo,
-            ],
-            untrackedScopes: [
-                d`d:/data`,
-                d`d:/app`,
-            ]
-        }
-    } : {};
+    let qTestContextInfoPath = undefined;
+    let untrackingCBPaths = {};
+    if (Environment.hasVariable("[Sdk.BuildXL]qtestContextInfo")){
+        const qTestContextInfoFile = Environment.getFileValue("[Sdk.BuildXL]qtestContextInfo");
+        qTestContextInfoPath = qTestContextInfoFile.path;
+        untrackingCBPaths =  {
+            unsafe: {
+                untrackedPaths: [
+                    qTestContextInfoFile,
+                ],
+                untrackedScopes: [
+                    d`d:/data`,
+                    d`d:/app`,
+                    Environment.hasVariable("QAUTHMATERIALROOT") ? d`${Environment.getDirectoryValue("QAUTHMATERIALROOT")}` : undefined,

```

Comment:
>Environment [](start = 20, length = 11)

I suspect this mgith fail if we have undefined in the list. Better would be:
```
...addIf(Environment.hasVariable("QAUTHMATERIALROOT"),
    Environment.getDirectoryValue("QAUTHMATERIALROOT")
);
```

Reply:
I just tried adding `undefined` as an untracked scope and it worked. 

While I agree with Danny's comment wholeheartedly, I suggest making this change in a separate PR.  The reason is that we still don't have a release candidate for next week, and we need this change to pass our canary validation, so I'd like to merge it as soon as we can.